src/stactools/goes_glm/stac.py

```diff
@@ -174,130 +175,161 @@ def create_item(
+            bbox = constants.ITEM_BBOX_WEST
+            geometry = constants.GEOMETRY_WEST
+        else:
+            bbox = None
+            geometry = None
+
+        centroid = {}
+        for key, var in dataset.variables.items():
+            if len(var.dimensions) != 0:
+                continue
+
+            ma = var[...]
+            if var.name == "lat_field_of_view":
+                centroid["lat"] = ma.tolist()
+            elif var.name == "lon_field_of_view":
+                centroid["lon"] = ma.tolist()
+            elif ma.count() > 0:
+                properties[f"goes-glm:{var.name}"] = ma.tolist()
+
+        item = Item(
+            stac_extensions=[
+                # todo: add extension again once released #12
+                # constants.GOES_EXTENSION,
+                constants.PROCESSING_EXTENSION,
+            ],
+            id=id,
+            properties=properties,
+            geometry=geometry,
+            bbox=bbox,
+            datetime=computed_datetime,
+            collection=collection,
+        )
+
+        proj = ProjectionExtension.ext(item, add_if_missing=True)
+        proj.epsg = constants.TARGET_CRS
+        if len(centroid) == 2:
+            proj.centroid = centroid
+
+        if not nogeoparquet:
+            target_folder = os.path.dirname(asset_href)
+            assets = parquet.convert(dataset, target_folder)
+            for key, asset_dict in assets.items():
+                asset = Asset.from_dict(asset_dict)
+                item.add_asset(key, asset)
+                TableExtension.ext(asset, add_if_missing=True)
+
+        if not nonetcdf:
+            # todo: replace with DataCube extension from PySTAC #16
+            item.stac_extensions.append(constants.DATACUBE_EXTENSION)
+            asset_dict = netcdf.create_asset(asset_href)
+            asset_dict["created"] = dataset.date_created
+            asset_dict["cube:dimensions"] = netcdf.to_cube_dimensions(dataset)
+            asset_dict["cube:variables"] = netcdf.to_cube_variables(dataset)
             asset = Asset.from_dict(asset_dict)
-            item.add_asset(key, asset)
-            TableExtension.ext(asset, add_if_missing=True)
+            item.add_asset(constants.NETCDF_KEY, asset)
 
-    if not nonetcdf:
-        # todo: replace with DataCube extension from PySTAC #16
-        item.stac_extensions.append(constants.DATACUBE_EXTENSION)
-        asset_dict = netcdf.create_asset(asset_href)
-        asset_dict["created"] = dataset.date_created
-        asset_dict["cube:dimensions"] = netcdf.to_cube_dimensions(dataset)
-        asset_dict["cube:variables"] = netcdf.to_cube_variables(dataset)
-        asset = Asset.from_dict(asset_dict)
-        item.add_asset(constants.NETCDF_KEY, asset)
-
-    return item
+        if missing_unsigned and not fixnetcdf:
+            for defect_var in defect_vars:
+                if defect_var in dataset.variables:
+                    del dataset.variables[defect_var]._Unsigned
```

Comment:
This makes me a little nervous since the `missing_unsigned` gets set to `True` regardless of whether all `defect_vars` are actually missing the `_Unsigned` attribute. However, per the validation doc, if one of the defective variables is missing the attribute, then all the defective variables should be missing the attribute. And the files we've tested thus far agree. So probably OK.

Reply:
I don't want anyone to feel nervous,so I made an improvement that keeps track of each individual variable. :-)

Reply:
Woops, thanks for the catch. Updated in the main branch.